### DISCUSSION TITLE
Adding a property to a collection query response

### DISCUSSION THREAD

**Discussion by PaulCombal** [api-platform/core#6326] [Q&A]
Hello,

I would like to add the total number of entities, regardless of the filters, to the reponse of a collection query (ex: GET /products).

[This documentation](https://api-platform.com/docs/core/serialization/#decorating-a-serializer-and-adding-extra-data) walks us through the steps to add a new key to an entity result, but how about a collection?

Using the same code than the above documentation (slightly updated for Symfony 7), I tried:
```
  App\Serializer\ApiCollectionNormalizer:
    decorates: 'api_platform.hydra.normalizer.collection'
```

But it gives me:
```
Typed property ApiPlatform\\Serializer\\AbstractCollectionNormalizer::$normalizer must not be accessed before initialization
```

As soon as I use `->normalize` on the decorated Normalizer. 

I must admit I don't exactly know what I'm doing.

Is there any way to achieve this properly?

Thank you

**Comment by MajesticGray** [comment]
same problem

**Comment by kreciszd** [comment]
Any solution for this ?
I want to edit collection data for example add new field to [hydra:view] object and I cannot do this with normalizer on collection
